### PR TITLE
Use annotations for validation, instead of code

### PR DESCRIPTION
**Pull request by antban**
Use annotations for validation, instead of code

Not to increase codebase and to improve readability

### CODE REVIEW

File: core-common/src/main/java/org/zalando/nakadi/annotations/validation/LabelKey.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+@Target({
+        ElementType.FIELD, ElementType.METHOD, ElementType.PARAMETER, ElementType.ANNOTATION_TYPE,
+        ElementType.TYPE_USE})
+@Retention(RetentionPolicy.RUNTIME)
+@NotEmpty(message = "Key cannot be empty")
+@Pattern(regexp = "^([a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*/)?[^/]*$",
+        message = "Key name should be an optional lowercase domain name with path")
+@Pattern(regexp = "^([-.a-z0-9]{0,253}/)?[^/]*$",
+        message = "Key prefix cannot be more than 253 characters.")
+@Pattern(regexp = "^([-.a-z0-9]+/)?[^/]{0,63}$",
```
Comment (burimshala) [inline core-common/src/main/java/org/zalando/nakadi/annotations/validation/LabelKey.java:21]:
```suggestion
@Pattern(regexp = "^(.+/)?[^/]{0,63}$",
```

File: core-common/src/main/java/org/zalando/nakadi/annotations/validation/LabelKey.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import javax.validation.Constraint;
+import javax.validation.Payload;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.Pattern;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target({
```
Comment (a1exsh) [inline core-common/src/main/java/org/zalando/nakadi/annotations/validation/LabelKey.java:12]: Is there difference from `AnnotationKey` validatior?

File: core-common/src/main/java/org/zalando/nakadi/domain/EventTypeBase.java [hunk 3]
```diff
@@ -67,11 +72,15 @@
 
     @Valid
     @Nullable
-    private ResourceAnnotations annotations;
+    private Map<
+            @AnnotationKey String,
```
Comment (a1exsh) [inline core-common/src/main/java/org/zalando/nakadi/domain/EventTypeBase.java:76]: Maybe `ValidAnnotationKey` for a name?

File: core-common/src/main/java/org/zalando/nakadi/annotations/validation/LabelKey.java (hunk 2 above)
Comment (antban) [reply core-common/src/main/java/org/zalando/nakadi/annotations/validation/LabelKey.java:10]: No

File: core-common/src/main/java/org/zalando/nakadi/domain/EventTypeBase.java (hunk 3 above)
Comment (antban) [reply core-common/src/main/java/org/zalando/nakadi/domain/EventTypeBase.java:76]: Looking at other annotation values: 
`Size`, `Max`, `Pattern`, `NotEmpty`, `Email`, I would say no...

Comment (a1exsh) [reply core-common/src/main/java/org/zalando/nakadi/domain/EventTypeBase.java:76]: Right.  It's good like that.

File: core-common/src/main/java/org/zalando/nakadi/annotations/validation/LabelValue.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target({
+        ElementType.FIELD, ElementType.METHOD, ElementType.PARAMETER, ElementType.ANNOTATION_TYPE,
+        ElementType.TYPE_USE})
+@Retention(RetentionPolicy.RUNTIME)
+@Size(max = 1000)
```
Comment (burimshala) [inline core-common/src/main/java/org/zalando/nakadi/annotations/validation/LabelValue.java:16]:
```suggestion
@Size(max = 63)
```

From [documentation](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/):

> Valid label value:
> must be 63 characters or less (can be empty),
> unless empty, must begin and end with an alphanumeric character ([a-z0-9A-Z]),
> could contain dashes (-), underscores (_), dots (.), and alphanumerics between.

**Comment by thomasabraham** [comment]
:+1:

**Comment by thomasabraham** [comment]
:+1:

**Comment by a1exsh** [comment]
:+1:

**Pull request merged**